paddle/fluid/framework/prune.cc

```diff
@@ -81,19 +83,47 @@ bool HasFalseTarget(const proto::OpDesc& op_desc) {
 }
 
 int GetSubBlockIndex(const proto::OpDesc& op_desc) {
+  // the block index >= 0, so -1 is used to indicate "NotFound".
   for (auto& attr : op_desc.attrs()) {
     if (attr.type() == proto::AttrType::BLOCK) {
-      PADDLE_ENFORCE(attr.has_block_idx());
+      PADDLE_ENFORCE_EQ(
+          attr.has_block_idx(), true,
+          platform::errors::NotFound("Attribute sub_block is empty"));
       return attr.block_idx();
     }
   }
   return -1;
 }
 
+void SetSubBlockIndex(proto::OpDesc* op_desc, int sub_idx) {
+  for (auto& attr : *op_desc->mutable_attrs()) {
+    if (attr.type() == proto::AttrType::BLOCK) {
+      PADDLE_ENFORCE_EQ(
+          attr.has_block_idx(), true,
+          platform::errors::NotFound("Attribute sub_block is empty"));
+      attr.set_block_idx(sub_idx);
+    }
+  }
+}
+
 bool HasSubBlock(const proto::OpDesc& op_desc) {
   return GetSubBlockIndex(op_desc) > 0;
 }
 
+int GetOpRole(const proto::OpDesc& op_desc) {
+  // the normal op role >= 0, so -1 is used to indicate "NotFound".
```

Comment:
the -> The.  
" the normal op role >= 0" may mean "The op_role >= 0 normally" ?

Reply:
Done, thanks.